File: Leetcode 3901-4000/3952/3952.java
// Leetcode 3952: Maximum Total Value of Covered Indices
// https://leetcode.com/problems/maximum-total-value-of-covered-indices/
// Solved on 21st of June, 2026
class Solution {
    /**
     * Calculates the maximum total value of covered indices.
     * 
     * @param nums The array of values.
     * @param s The string indicating covered indices.
     * @return The maximum total value of covered indices.
     */
    public long maxTotal(int[] nums, String s) {
        long totalSum = 0;
        int n = nums.length;
        int i = 0;
        while (i < n) {
            if (s.charAt(i) == '1') {
                int l = i;
                while (i < n && s.charAt(i) == '1') {
                    i++;
                }
                int r = i - 1;
                if (l == 0) {
                    for (int j = 0; j <= r; j++) {
                        totalSum += nums[j];
                    }
                } else {
                    int minVal = nums[l - 1];
                    long blockSum = nums[l - 1];
                    for (int j = l; j <= r; j++) {
                        blockSum += nums[j];
                        if (nums[j] < minVal) {
                            minVal = nums[j];
                        }
                    }
                    totalSum += (blockSum - minVal);
                }
            } else {
                i++;
            }
        }
        return totalSum;
    }
}

File: Leetcode 3901-4000/3952/3952.py
# Leetcode 3952: Maximum Total Value of Covered Indices
# https://leetcode.com/problems/maximum-total-value-of-covered-indices/
# Solved on 21st of June, 2026
class Solution:
    def maxTotal(self, nums: list[int], s: str) -> int:
        """
        Calculates the maximum total value of covered indices.
        
        :param nums: The array of values.
        :param s: The string indicating covered indices.
        :return: The maximum total value of covered indices.
        """
        numElements = len(nums)

        totalValue = 0
        currentIndex = 0

        while currentIndex < numElements:
            if s[currentIndex] == '1':
                leftIndex = currentIndex
                while currentIndex < numElements and s[currentIndex] == '1':
                    currentIndex += 1
                
                rightIndex = currentIndex - 1
                if leftIndex == 0:
                    totalValue += sum(nums[0 : rightIndex + 1])
                else:
                    totalValue += sum(nums[leftIndex - 1 : rightIndex + 1]) - min(nums[leftIndex - 1 : rightIndex + 1])
            else:
                currentIndex += 1

        return totalValue